control_client/src/main.c

```diff
@@ -122,30 +126,58 @@ int main(int argc, char **argv) {
         }
 
         for (unsigned int i = 0; i < sizeof(commands) / sizeof(commands[0]); i++) {
-            
+
             if (commands[i].key == key) {
                 header_p hdr = {.type = TYPE_CNTRL, .subtype = commands[i].subtype}; // create header
+                struct iovec iov[2];
+                iov[0].iov_base = &hdr;
+                iov[0].iov_len = sizeof(hdr);
 
                 // check what type of command it is
                 switch (commands[i].subtype) {
                 case CNTRL_ACT_REQ:
-                    switch_t *actuator = commands[i].priv; // get the actuator data
-                    actuator->state = !actuator->state;    // flip the state
+                    switch_t *actuator = commands[i].priv;                                  // get the actuator data
+                    actuator->state = !actuator->state;                                     // flip the state
                     act_req_p act_req = {.id = actuator->act_id, .state = actuator->state}; // Create message
 
-                    pad_send(&pad, &hdr, sizeof(hdr));
-                    pad_send(&pad, &act_req, sizeof(act_req));
+                    iov[1].iov_base = &act_req;
+                    iov[1].iov_len = sizeof(act_req);
+                    pad_send(&pad, iov);
+
+                    act_ack_p act_ack;
+                    err = pad_receive(&pad, &act_ack, sizeof(act_ack));
+
+                    if (err == -1) {
+                        fprintf(stderr, "Server did not aknowledge actuator request with id %d", actuator->act_id);
```

Comment:
Nitpicky but add newlines to your error printing because otherwise the messages will be a pain to read. Applies to all fprintf calls.